src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -90,7 +90,7 @@ private void WriteShExecutionScript(string shExecutionTemplate, string outputPat
             foreach (string runCommand in TestCommands)
             {
                 testRunCommands.Append($"{runCommand}\n");
-                testRunEchoes.Append($"echo {runCommand}\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
+                testRunEchoes.Append($"echo \"{runCommand}\"\n".Replace("(", "").Replace(")", ""));  // Remove parentheses from echo command to avoid errors on Linux
```

Comment:
I checked on Ubuntu and double quotes in the command itself whether it is escaped (`\"`) or not (`""`), does not get affected if you wrap it with double quotes. Do you know of any case which could be problematic? @MattGal